crates/turborepo-lib/src/package_manager.rs

```diff
@@ -41,17 +48,28 @@ impl From<Workspaces> for Vec<String> {
     }
 }
 
+#[derive(Debug, Serialize, PartialEq, Eq, Clone)]
+#[serde(rename_all = "lowercase")]
 pub enum PackageManager {
```

Comment:
Yeah, so I was debating this because on one hand the Go code made it easy to add a single package manager to the code, but it also made it annoying to navigate and read the code because of that split. It also meant that we did redundant work like shelling out to yarn multiple times. What I could do is break it up into a `PnpmDetector` that then returns either a `PackageManager::Pnpm` or a `PackageManager::Pnpm6`, and then do `detectors.par_iter().collect()`.

I do want to bias towards readability though because realistically we're not going to be adding a package manager very frequently (famous last words).

Reply:
Bun and Deno called.

Reply:
That's still only two. If we're getting a new package manager every couple weeks, like say we start doing multi-language and yarn decides to release yarn squash which breaks compatibility with yarn berry, and pnpm decides that actually, yaml is evil and everybody should be using toml, then sure, we can refactor into a more abstract version, but until then, I'd vote that we stick to as little abstraction as possible.

With all that said, I think the detector is a good compromise because it composes well but avoids the redundant work

Reply:
I'm not sure if this is better though. If someone were to add a new package manager, they'd have to create a new detector *and* edit the package_manager/mod.rs file. I could try to separate the logic even more, so that there are separate modules for each package manager, but I feel like that'd end up more convoluted. Without the detectors, while the package manager file is a monolith, adding a new package manager requires touching just that one file, which frankly isn't even that long.